prime-router/docs/design/proposals/translation-template-test-plan.md

```diff
@@ -0,0 +1,57 @@
+# Translation Template Test Plan
+
+## Problem Statement
+As a Reportstream engineer, I want tests that verify that the changes I make to translations will not break essential 
+rules set forth by the spec or our users. 
+
+Currently, we use integration tests for this purpose, but these do not give us the ability to easily tell which thing 
+broke/changed or even really what is specifically being tested.
+
+## Considerations
+- We not only want to check the rules set forth in the spec, but the specific ones created for senders and receivers.
+
+## Proposal on *what* to test
+We should write tests when:
+- The spec specifies a condition.
+- There is a requirement specified by a sender or receiver. This means that there needs to be a test for every rule in
+  a sender or receiver transform, This includes the default transform since these are ReportStream specific rules we
+  are implementing
+  - This can be done by comparing segments rather than field by field.
+- We have created an extension not specified in the spec.
+- We also want to test the losslessness of an HL7 message being transformed into a FHIR message and then back to an
+HL7 message. Losslessness beign defined as the data being in the same place it was when originally passed in.
+
+## Options for *how* to test 
+This proposal is not choosing a way to test, just documenting the concerns brought up by the team in the process of 
+making the proposal about the different types of tests that we could use.  
+1. Continue with integration tests
+    - These would be specific to the field they are trying to test rather than a cluster of fields
+    - Named after the field they are trying to test
+    - We can leverage the new diff tool to check if the output is what we expect.
+
+   - Pros:
+     - Setup already exists, works, and is tested 
+   - Cons:
+     - The file to add/updates/remove tests would become difficult to read and find tests in because of the sheer number
+     - Many files would need to be added
+     - Would need to revamp how tests are created to mitigate the other cons. 
+2. Unit Tests
+    - We can leverage the new diff tool to check if the output is what we expect.
+
+    - Pros:
+        - Easier to read, can have a specific test for each field and have it cleanly labeled rather than a list of
+files that you would need to scroll through.
+        - If we use Strings instead of files, these will all be kept in one place.
+        - Can break tests out into one file per transform file for further organization.
+    - Cons: 
+        - Many files would need to be added or there will be a lot of lengthy strings within the tests.
+        - Odd fit since this is to test transform changes rather than code changes.
+        - Would cause the jar to not build which seems like a hefty and unwanted consequence for just a misaligned 
```

Comment:
This one sounds like such a big Con that it almost takes this option out of the running